discord-bot/src/register-api-key.ts

```diff
@@ -0,0 +1,37 @@
+import { Message } from 'discord.js'
+import { writeApiKeyToDiscordUserId } from './storage.js'
+
+export const registerApiKey = async (message: Message) => {
+  const key = message.content.trim()
+  if (key.length !== 36) {
+    await message.reply(
+      'API key should be 36 characters long, find it at https://manifold.markets/my-api-key'
+    )
+    return
+  }
+  let me
+  await message.reply({
+    content: 'Checking API key...',
+  })
+  let failed = false
+  try {
+    me = await fetch('https://manifold.markets/api/v0/me', {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Key ${key}`,
+      },
```

Comment:
It seems like this should be in parallel with creating the "Checking API key..." message above.